crates/router/src/connector/checkout/transformers.rs

```diff
@@ -537,10 +569,16 @@ impl ForeignFrom<(CheckoutPaymentStatus, Option<Balances>)> for enums::AttemptSt
                     Self::Authorized
                 }
             }
-            CheckoutPaymentStatus::Captured => Self::Charged,
-            CheckoutPaymentStatus::Declined => Self::Failure,
+            CheckoutPaymentStatus::Captured
+            | CheckoutPaymentStatus::PartiallyRefunded
+            | CheckoutPaymentStatus::Refunded => Self::Charged,
+            CheckoutPaymentStatus::PartiallyCaptured => Self::PartialCharged,
+            CheckoutPaymentStatus::Declined | CheckoutPaymentStatus::Expired => Self::Failure,
             CheckoutPaymentStatus::Pending => Self::AuthenticationPending,
-            CheckoutPaymentStatus::CardVerified => Self::Pending,
+            CheckoutPaymentStatus::CardVerified | CheckoutPaymentStatus::RetryScheduled => {
+                Self::Pending
+            }
+            CheckoutPaymentStatus::Voided | CheckoutPaymentStatus::Canceled => Self::Voided,
```

Comment:
When do we get this Canceled status? if this is during authentication then should be mapped to payment failure